plugin/code_actions.py

```diff
@@ -323,23 +351,39 @@ def on_selection_modified_async(self) -> None:
                 lambda: self._stored_region == current_region, async_thread=True)
 
     def fire_request(self, current_region: sublime.Region) -> None:
-        self._actions = []
-        actions_manager.request(self.view, self.handle_responses, current_region.begin())
+        self._actions_by_config = {}
+        view = self.view
+        stored_range = Range(Point(*view.rowcol(self._stored_region.begin())),
+                             Point(*view.rowcol(self._stored_region.end())))
+        diagnostics_by_config = filter_by_range(view_diagnostics(view), stored_range)
+        actions_manager.request_with_optional_diagnostics(view, diagnostics_by_config, self.handle_responses)
 
     def handle_responses(self, responses: CodeActionsByConfigName) -> None:
-        for _, items in responses.items():
-            self._actions.extend(items)
-        if len(self._actions) > 0:
-            self.show_bulb()
+        self._actions_by_config = responses
+        if any(self._actions_by_config.values()):
+            self.show_annotations()
+            if settings.show_code_actions_bulb:
+                self.show_bulb()
 
     def show_bulb(self) -> None:
-        region = self.view.sel()[0]
         flags = sublime.DRAW_NO_FILL | sublime.DRAW_NO_OUTLINE
-        self.view.add_regions('lsp_bulb', [region], 'markup.changed', 'Packages/LSP/icons/lightbulb.png', flags)
+        self.view.add_regions('lsp_bulb', [self._stored_region], 'markup.changed',
+                              'Packages/LSP/icons/lightbulb.png', flags)
 
     def hide_bulb(self) -> None:
         self.view.erase_regions('lsp_bulb')
 
+    def show_annotations(self) -> None:
+        flags = sublime.DRAW_NO_FILL | sublime.DRAW_NO_OUTLINE
+        actions_count = sum(map(len, self._actions_by_config.values()))
+        code_actions_link = make_link('subl:lsp_code_actions', '{} code action(s)'.format(actions_count))
+        self.view.add_regions('lsp_action_annottations', [self._stored_region], flags=flags,
+                              annotations=["<div class=\"actions\">{}</div>".format(code_actions_link)],
+                              annotation_color='yellow')
```

Comment:
Can we use `var(--yellowish)` instead? Or `region.yellowish` ?

Reply:
None of those work.